### PR TITLE
#608 Add localized computed method to subscribe to locale store changes

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-white/608 |
| Decisions | Add localized computed method to subscribe to locale store changes. This does the same as `locale` method but is reactive to `locale` changes in the store. |

### CODE REVIEW

**Review by gcandal — APPROVED** [review]

File: vue/plugins/locale.js [hunk 1]
```diff
@@ -1,10 +1,32 @@
...
+        const _locale = function(value, defaultValue, locale, fallback = true) {
             if (!value) return defaultValue || "";
             value = value.toString();
             defaultValue = defaultValue !== undefined ? defaultValue : value;
             return localePlugin.toLocale(value, defaultValue, locale, fallback);
+        };
+
+        Vue.mixin({
+            computed: {
+                localeReactive() {
```
Comment (gcandal) [inline vue/plugins/locale.js:12]: locale

File: vue/plugins/locale.js [hunk 2]
```diff
@@ -1,10 +1,32 @@
...
+                    return (value, defaultValue, locale, fallback = true) => {
+                        return (
+                            (this.$store && this.$store.state.locale || "unset") &&
+                            this.$options.filters.locale(value, defaultValue, locale, fallback)
+                        );
+                    };
+                }
+            },
+            methods: {
+                locale(value, defaultValue, locale, fallback = true) {
```
Comment (gcandal) [inline vue/plugins/locale.js:22]: delete (check retail)

File: vue/plugins/locale.js [hunk 3]
```diff
@@ -1,10 +1,32 @@
...
             defaultValue = defaultValue !== undefined ? defaultValue : value;
             return localePlugin.toLocale(value, defaultValue, locale, fallback);
+        };
+
+        Vue.mixin({
+            computed: {
+                localeReactive() {
+                    return (value, defaultValue, locale, fallback = true) => {
+                        return (
+                            (this.$store && this.$store.state.locale || "unset") &&
```
Comment (gcandal) [inline vue/plugins/locale.js:15]: const abc = locale | store.state.local | "en_us"

**Review by gcandal — CHANGES_REQUESTED** [review]

File: vue/test/mocks/mixins/locale.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export const localeMockMixin = {
+    computed: {
+        locale() {
+            return (value, defaultValue) => value || defaultValue;
+        }
+    },
+    methods: {
+        locale(value, defaultValue) {
```
Comment (joamag) [inline vue/test/mocks/mixins/locale.js:8]: @gcandal having locale method plus locale computed, works?

Comment (gcandal) [reply vue/test/mocks/mixins/locale.js]: It works but doesn't make sense. Fixed.

**Review by joamag — APPROVED** [review]

**Pull request merged**